js/ui/drop_down_editor/ui.drop_down_editor.js

```diff
@@ -617,7 +617,30 @@ const DropDownEditor = TextBox.inherit({
     },
 
     _popupPositionedHandler: function(e) {
-        e.position && this._popup.$overlayContent().toggleClass(DROP_DOWN_EDITOR_OVERLAY_FLIPPED, e.position.v.flip);
+        const $popupOverlayContent = this._popup.$overlayContent();
+        const isOverlayFlipped = e.position.v.flip;
+        const labelMode = this.option('labelMode');
+        const stylingMode = this.option('stylingMode');
+
+        const isNeedMoved = (labelMode, stylingMode) => (
+            ((labelMode === 'static' && stylingMode === 'outlined') ||
+            (labelMode === 'floating' && stylingMode === 'outlined') ||
+            (labelMode === 'static' && stylingMode === 'underlined') ||
+            (labelMode === 'floating' && stylingMode === 'underlined')) &&
+            isOverlayFlipped && this._label.isVisible()
+        );
+
+        if(e.position) {
+            $popupOverlayContent.toggleClass(DROP_DOWN_EDITOR_OVERLAY_FLIPPED, isOverlayFlipped);
+        }
+
+        if(isNeedMoved(labelMode, stylingMode)) {
```

Comment:
Declare the function inside other function it's not good idea. Will better if it be a method of our component that contained correspond logic.